app/src/main/java/io/zenandroid/onlinego/data/model/local/Puzzle.kt

```diff
@@ -0,0 +1,62 @@
+package io.zenandroid.onlinego.data.model.local
+
+import androidx.room.ColumnInfo
+import androidx.room.Embedded
+import androidx.room.Entity
+import androidx.room.Ignore
+import androidx.room.PrimaryKey
+import io.zenandroid.onlinego.data.model.local.Player
+import io.zenandroid.onlinego.data.model.local.PuzzleCollection
+import io.zenandroid.onlinego.data.model.ogs.OGSPlayer
+import io.zenandroid.onlinego.data.model.ogs.OGSPuzzle
+import io.zenandroid.onlinego.data.model.ogs.OGSPuzzle.PuzzleData
+import io.zenandroid.onlinego.data.model.ogs.OGSPuzzleCollection
+import java.time.Instant
+
+@Entity
+data class Puzzle(
+    @PrimaryKey var id: Long,
```

Comment:
All of the fields of the model need to be `val` and not `var`. We should also mark the data classes as `@Immutable`, particularly if the contain `List` objects. This is not just pedantic, it's because of how Compose works: If the state is immutable, then many recompositions can be skipped and performance drastically improves. See Google's recommendations here https://developer.android.com/jetpack/compose/performance/stability

And yes, I know I haven't changed it everywhere myself.